From 7de33e177b5c37bbeb6146a85f02aa1ab89a3cff Mon Sep 17 00:00:00 2001
From: Fei Jiang <fei.jiang@intel.com>
Date: Tue, 12 Jun 2018 11:15:21 +0800
Subject: [PATCH 1559/2328] drm/i915/gvt: fix i915 boot issue when gvt is not
 enabled

Previous plane scaling patch didn't consider following case: ACRN_GVT is
built in kernel, while didn't enable SOS gvt in kernel cmdline. In such
case, there was kernel boot crash issue, add gvt check to avoid the problem.

Signed-off-by: Fei Jiang <fei.jiang@intel.com>
---
 drivers/gpu/drm/i915/intel_display.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/gpu/drm/i915/intel_display.c b/drivers/gpu/drm/i915/intel_display.c
index 096aa8f..6ea01a7 100644
--- a/drivers/gpu/drm/i915/intel_display.c
+++ b/drivers/gpu/drm/i915/intel_display.c
@@ -13642,6 +13642,7 @@ static void intel_crtc_init_scalers(struct intel_crtc *crtc,
 		scaler->owned = 1;
 #if IS_ENABLED(CONFIG_DRM_I915_GVT)
 		if (!intel_vgpu_active(dev_priv) &&
+		    intel_gvt_active(dev_priv) &&
 		    dev_priv->gvt->pipe_info[crtc->pipe].scaler_owner[i] != 0)
 			scaler->owned = 0;
 #endif
-- 
2.7.4

